packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -714,48 +895,41 @@ final class SignInStateMachine
         clientMetadata: event?.clientMetadata,
       );
 
+      if (_enableMfaType case final enableMfaType?) {
+        await cognitoIdentityProvider.setMfaSettings(
+          accessToken: accessToken,
+          sms: enableMfaType == MfaType.sms ? MfaPreference.enabled : null,
+          totp: enableMfaType == MfaType.totp ? MfaPreference.enabled : null,
+        );
+      }
+
       return SignInState.success(_user.build().authUser);
     }
 
     await _updateUser(_challengeParameters);
 
+    // Configure TOTP authentication if allowed.
+    if (_allowedMfaTypes case final allowedMfaTypes?
+        when _challengeParameters
+            .containsKey(CognitoConstants.challengeParamMfasCanSetup)) {
+      if (allowedMfaTypes.contains(MfaType.totp)) {
+        _totpSetupResult ??= await associateSoftwareToken();
+      } else {
+        throw const SoftwareTokenMfaNotFoundException(
+          'Cannot enable SMS MFA and TOTP MFA is not allowed',
+          recoverySuggestion:
+              'Contact an administrator to enable SMS MFA or allow TOTP MFA',
+        );
+      }
+    }
+
```

Comment:
```suggestion
    // Configure TOTP authentication if allowed.
    if (_allowedMfaTypes case final allowedMfaTypes?
        when _challengeParameters
            .containsKey(CognitoConstants.challengeParamMfasCanSetup)) {
      if (!allowedMfaTypes.contains(MfaType.totp)) {
         throw const SoftwareTokenMfaNotFoundException(
          'Cannot enable SMS MFA and TOTP MFA is not allowed',
          recoverySuggestion:
              'Contact an administrator to enable SMS MFA or allow TOTP MFA',
        );
      }
      _totpSetupResult ??= await associateSoftwareToken();
    }

```
nit: could use a guard clause here